src/Stratis.Bitcoin.Features.BlockStore/BlockStoreLoop.cs

```diff
@@ -219,7 +219,7 @@ public void AddToPending(BlockPair blockPair)
         /// </summary>
         internal void ShutDown()
         {
-            this.asyncLoop.Dispose();
+            this.asyncLoop?.Dispose(); // null check in case loop was not started.
```

Comment:
Was there a situation where this threw an exception @Neurosploit ? Just asking out of curiosity.

Also, the comment isn't needed ;)

Reply:
@fassadlr Whilst I was working on the checkpoints PR I made some changes and an integration test failed which led me to find this scenario.